dvc/repo/experiments/queue/base.py

```diff
@@ -323,12 +323,12 @@ def _stash_exp(  # noqa: C901
         .. _Hydra Override:
             https://hydra.cc/docs/next/advanced/override_grammar/basic/
         """
-        with self.scm.detach_head(client="dvc") as orig_head:
```

Comment:
@pmrowla Was there a reason to detach head _before_ stash workspace?

Reply:
No, but when we were not reinstating index the order doesn't matter. With the index reinstate, your change is correct (since we need to pop the stash after the `reset()` when we exit the `detach_head` context)